pkg/proc/threads.go

```diff
@@ -105,320 +101,6 @@ func topframe(g *G, thread Thread) (Stackframe, Stackframe, error) {
 	}
 }
 
-// ErrNoSourceForPC is returned when the given address
-// does not correspond with a source file location.
-type ErrNoSourceForPC struct {
-	pc uint64
-}
-
-func (err *ErrNoSourceForPC) Error() string {
-	return fmt.Sprintf("no source for PC %#x", err.pc)
-}
-
-// Set breakpoints at every line, and the return address. Also look for
-// a deferred function and set a breakpoint there too.
-// If stepInto is true it will also set breakpoints inside all
-// functions called on the current source line, for non-absolute CALLs
-// a breakpoint of kind StepBreakpoint is set on the CALL instruction,
-// Continue will take care of setting a breakpoint to the destination
-// once the CALL is reached.
-//
-// Regardless of stepInto the following breakpoints will be set:
-// - a breakpoint on the first deferred function with NextDeferBreakpoint
-//   kind, the list of all the addresses to deferreturn calls in this function
-//   and condition checking that we remain on the same goroutine
-// - a breakpoint on each line of the function, with a condition checking
-//   that we stay on the same stack frame and goroutine.
-// - a breakpoint on the return address of the function, with a condition
-//   checking that we move to the previous stack frame and stay on the same
-//   goroutine.
-//
-// The breakpoint on the return address is *not* set if the current frame is
-// an inlined call. For inlined calls topframe.Current.Fn is the function
-// where the inlining happened and the second set of breakpoints will also
-// cover the "return address".
-//
-// If inlinedStepOut is true this function implements the StepOut operation
-// for an inlined function call. Everything works the same as normal except
-// when removing instructions belonging to inlined calls we also remove all
-// instructions belonging to the current inlined call.
-func next(dbp Process, stepInto, inlinedStepOut bool) error {
```

Comment:
I'd like to leave this here, even if it doesn't make sense (and hasn't for a long time), to preserve git history.

Reply:
On second thought this is a really bad place for this so moving it is probably warranted. How about moving all the resume functions into proc.go?

Reply:
I understand the want for preserving git history, but I also do not like scattering methods for the same type across different files. Tracking down the history wouldn't be impossible either, a `git blame` would reveal the move and commit SHA, which could be checked out and then `git blame`ed again.

Reply:
But if they were all in proc.go they would all be together.

Reply:
Area you suggesting we get rid of `proc/target.go` and then move the struct definition, existing methods etc over to `proc/proc.go`?

Reply:
No, I'm suggesting that we move next and it's helper functions in proc.go along with the other functions that resume process execution. Go doesn't restrict methods of a type to live all in the same file, unlike java, so no reason to move all the methods of target in the same file and target.go does not currently contain any target-resuming functions.

Reply:
Right, I realize that, but from a code organization standpoint I'm not a fan of having multiple files define methods on the same type. I think it hurts discoverability and I don't want to make code organization decisions based off of the benefit of git history. I look at the file much more often than I run `git blame` on it, so I'd rather optimize for development.

For a new user to the project, you would not expect methods of `Target` to exist in both `proc/target.go` and `proc/proc.go`.

Reply:
Personally I prefer methods to be grouped by topic rather than by receiver. Grep and gopls exist to find where a method is located and I'd rather have it surrounded by other methods that related to it than with methods that happen to be on the same type.

For example (*Variable).pointerToVariable could be defined inside variables.go instead of eval.go but the latter is better since eval.go is the only place where it's used.

I do look at the file more often than I use git blame but having to grep for a specific function isn't too much truble whereas searching the history of moved function with git blame is more time consuming when I have to do it.

Reply:
I still think the tooling should help us here, not the other way around.

Also, `git blame` has a solution for the predicament: https://git-scm.com/docs/git-blame (see `-C` and `-M`).

Reply:
I'd rather not depend on git blame working correctly on complex cases and even if it does I stand by my other arguments.